### ISSUE TITLE
Qt5 build libGL missing error 

### ISSUE THREAD

**Issue by machinekoder** [nezticle/RaspberryPi-BuildRoot#17]
I just built bsquask and tried to build the Qt5 libraries. I got the following 
error:
/opt/bsquask/host/opt/ext-toolchain/bin/../lib/gcc/arm-raspberrypi-linux-
gnueabi/4.6.3/../../../../arm-raspberrypi-linux-gnueabi/bin/ld: cannot find -
lGL

I used this configure:
./configure -prefix /usr -hostprefix $BSQUASK_HOST_DIR/usr -release -device pi -
make libs \
-device-option CROSS_COMPILE=$BSQUASK_HOST_DIR/usr/bin/arm-raspberrypi-linux-
gnueabi- \
-device-option DISTRO=bsquask -sysroot $BSQUASK_STAGING_DIR -no-neon -
opensource -confirm-license

adding -opengl es2 also ends up in an error.

**Comment by nezticle** [comment]
What is the error you get when you add -opengl es2?  There is no libGL package provided, as this is for desktop OpenGL, and Raspberry Pi provides OpenGL ES2.  Are you trying to manually build Qt 5 from a specific SHA1, or building using the version provided?

**Comment by machinekoder** [comment]
Here is the full configure output: 

This is the Qt Open Source Edition.

You are licensed to use this software under the terms of
the Lesser GNU General Public License (LGPL) versions 2.1.

You have already accepted the terms of the  license.

<srcbase> = /home/alexander/bin/qt5/qtbase 
<outbase> = /home/alexander/bin/qt5/qtbase 
Creating qmake. Please wait...
g++ -c -o qlibraryinfo.o -m64 -pipe -DQMAKE_OPENSOURCE_EDITION -g  -I/home/alexander/bin/qt5/qtbase/qmake -I/home/alexander/bin/qt5/qtbase/qmake/library -I/home/alexander/bin/qt5/qtbase/qmake/generators -I/home/alexander/bin/qt5/qtbase/qmake/generators/unix -I/home/alexander/bin/qt5/qtbase/qmake/generators/win32 -I/home/alexander/bin/qt5/qtbase/qmake/generators/mac -I/home/alexander/bin/qt5/qtbase/qmake/generators/integrity -I/home/alexander/bin/qt5/qtbase/include -I/home/alexander/bin/qt5/qtbase/include/QtCore -I/home/alexander/bin/qt5/qtbase/include/QtCore/5.0.0 -I/home/alexander/bin/qt5/qtbase/include/QtCore/5.0.0/QtCore -I/home/alexander/bin/qt5/qtbase/src/corelib/global -DHAVE_QCONFIG_CPP -I/home/alexander/bin/qt5/qtbase/mkspecs/linux-g++-64 -I/home/alexander/bin/qt5/qtbase/tools/shared -DQT_BUILD_QMAKE -DQT_BOOTSTRAPPED -DPROEVALUATOR_FULL -DPROEVALUATOR_DEBUG -DQT_NO_TEXTCODEC -DQT_NO_UNICODETABLES -DQT_NO_COMPONENT -DQT_NO_COMPRESS -DQT_NO_THREAD -DQT_NO_QOBJECT -DQT_NO_GEOM_VARIANT -DQT_NO_DATASTREAM /home/alexander/bin/qt5/qtbase/src/corelib/global/qlibraryinfo.cpp
g++ -o "/home/alexander/bin/qt5/qtbase/bin/qmake" project.o option.o property.o main.o ioutils.o proitems.o qmakeglobals.o qmakeparser.o qmakeevaluator.o qmakebuiltins.o makefile.o unixmake2.o unixmake.o mingw_make.o winmakefile.o projectgenerator.o meta.o makefiledeps.o metamakefile.o xmloutput.o pbuilder_pbx.o msvc_vcproj.o msvc_vcxproj.o msvc_nmake.o msvc_objectmodel.o msbuild_objectmodel.o gbuild.o cesdkhandler.o qtextcodec.o qutfcodec.o qstring.o qstringbuilder.o qtextstream.o qiodevice.o qmalloc.o qglobal.o qarraydata.o qbytearray.o qbytearraymatcher.o qdatastream.o qbuffer.o qlist.o qfiledevice.o qfile.o qfilesystementry.o qfilesystemengine_unix.o qfilesystemengine.o qfilesystemiterator_unix.o qfsfileengine_unix.o qfsfileengine.o qfsfileengine_iterator.o qregexp.o qvector.o qbitarray.o qdir.o qdiriterator.o quuid.o qhash.o qfileinfo.o qdatetime.o qstringlist.o qabstractfileengine.o qtemporaryfile.o qmap.o qmetatype.o qsettings.o qsystemerror.o qlibraryinfo.o qvariant.o qvsnprintf.o qlocale.o qlocale_tools.o qlocale_unix.o qlinkedlist.o qnumeric.o qcryptographichash.o qxmlstream.o qxmlutils.o qlogging.o  -m64
Note: PKG_CONFIG_LIBDIR automatically set to /opt/bsquask/staging/usr/lib/pkgconfig:/opt/bsquask/staging/usr/share/pkgconfig:/opt/bsquask/staging/usr/lib/arm-raspberrypi-linux-gnueabi/pkgconfig
Note: PKG_CONFIG_SYSROOT_DIR automatically set to /opt/bsquask/staging

Could not determine the host architecture!
Turn on verbose messaging (-v) to see the final report.
The OpenGL ES 2.0 functionality test failed!
 You might need to modify the include and library search paths by editing
 QMAKE_INCDIR_OPENGL_ES2, QMAKE_LIBDIR_OPENGL_ES2 and QMAKE_LIBS_OPENGL_ES2 in
 /home/alexander/bin/qt5/qtbase/mkspecs/devices/linux-rasp-pi-g++.

I just made a full build and then followed the steps in this walkthrough: https://github.com/nezticle/RaspberryPi-BuildRoot/wiki/Walkthrough

**Comment by machinekoder** [comment]
Here the verbose log: https://dl.dropbox.com/u/50861330/log.txt

**Comment by machinekoder** [comment]
Do you have any suggestions?

**Comment by nezticle** [comment]
My guess would be that you don't have the VideoCore files in your BuildRoot sysroot when you are trying to build qtbase.  You can check for this by running "make VideoCore" in your BuildRoot build folder (so probably in your case  /opt/bsquask )

**Comment by machinekoder** [comment]
seems to be fine:
# make VideoCore
make -C /home/alexander/bin/RPi-Buildroot/BuildRoot O=/opt/bsquask/. VideoCore
make[1]: Nothing to be done for `VideoCore'.

**Comment by nezticle** [comment]
I updated the Qt5 packages today and they are still building fine.  There is at least once difference between upstream qtbase and my packages, and that is this patch:
https://raw.github.com/nezticle/RaspberryPi-BuildRoot/master/package/qt5/qtbase/qtbase.Revert-make-configure-set-up-QMAKE_DEFAULT_-INC-LIB-.patch

I wouldn't think that would affect the opengl test though.  The second thing to look at is the latest build instructions, which i've defined in the package here:
https://raw.github.com/nezticle/RaspberryPi-BuildRoot/master/package/qt5/qtbase/qtbase.mk

And finally, I'm not entirely sure how much you monkeyed around with the default config.  With buildroot its quite easy destroy your environment if you uncheck the wrong thing.  My fork is even more sensitive to this as I have made certain assumptions about dependencies.  If you've tried to hack and slash lots of stuff from the config, and don't know the repercussions of each check box, then your going to have a bad time.
